ocaml/xapi/xapi_observer_components.mli

```diff
@@ -0,0 +1,17 @@
+module Component : sig
+  type t = Xapi | Xenopsd | Xapi_clusterd | SMApi [@@deriving ord]
+
+  val all : t list
+
+  val to_string : t -> string
+
+  val of_string : string -> t
```

Comment:
Might want to add that this may fail with an exception.

Reply:
Added.